File: PS 5/test_crud.py
from selenium import webdriver
from selenium.webdriver.edge.options import Options
from selenium.common.exceptions import WebDriverException
import time

# Setup Edge options
options = Options()
options.add_experimental_option("excludeSwitches", ["enable-logging"])
options.set_capability("ms:loggingPrefs", {"browser": "ALL"})

# Start driver
driver = webdriver.Edge(options=options)

# Open your HTML file
driver.get("D:\\AISSMS IOIT\\Third Year\\6th Semester\\DEVOPS\\DevOps EndSem\\PS 5\\index.html")

print("---------------------------")
print("Test Started")
print("---------------------------")

try:
    while True:
        try:
            # ✅ Check if browser is still open
            if len(driver.window_handles) == 0:
                raise WebDriverException("Browser closed")

            # Get browser logs
            logs = driver.get_log("browser")

            for log in logs:
                message = log["message"]

                if "CREATED" in message:
                    print("TC-1 | Add Student    | PASS")
                elif "UPDATED" in message:
                    print("TC-2 | Edit Student   | PASS")
                elif "DELETED" in message:
                    print("TC-3 | Delete Student | PASS")

            time.sleep(1)

        except WebDriverException:
            # ✅ This will now trigger when browser closes
            print("\n---------------------------")
            print("TC-4 | Browser Close | PASS")
            print("---------------------------")
            break

except KeyboardInterrupt:
    print("\nManual Stop Detected")

finally:
    try:
        driver.quit()
    except:
        pass

File: Using JS/test_crud.js
const puppeteer = require("puppeteer-core");
const path = require("path");
const fs = require("fs");
 
// Path to your index.html
const filePath =
  "file://" +
  path.resolve("D:/AISSMS IOIT/Third Year/6th Semester/DEVOPS/DevOps EndSem/PS 5/index.html");
 
// Common Edge installation paths on Windows
const edgePaths = [
  "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe",
  "C:\\Program Files\\Microsoft\\Edge\\Application\\msedge.exe",
];
 
const edgeExecutable = edgePaths.find(fs.existsSync) || edgePaths[0];
 
(async () => {
  const browser = await puppeteer.launch({
    executablePath: edgeExecutable,
    headless: false,
    channel: "msedge",
    args: [
      "--no-sandbox",
      "--disable-setuid-sandbox",
      "--remote-debugging-port=9222",
      "--disable-dev-shm-usage",
      "--disable-extensions",
      "--no-first-run",
      "--no-default-browser-check",
    ],
    timeout: 60000,
  });
 
  const page = await browser.newPage();
 
  // Listen for browser console logs
  page.on("console", (msg) => {
    const text = msg.text();
 
    if (text.includes("CREATED")) {
      console.log("TC-1 | Add Student    | PASS");
    } else if (text.includes("UPDATED")) {
      console.log("TC-2 | Edit Student   | PASS");
    } else if (text.includes("DELETED")) {
      console.log("TC-3 | Delete Student | PASS");
    }
  });
 
  // Detect browser/page close
  page.on("close", () => {
    console.log("\n---------------------------");
    console.log("TC-4 | Browser Close | PASS");
    console.log("---------------------------");
    process.exit(0);
  });
 
  await page.goto(filePath);
 
  console.log("---------------------------");
  console.log("Test Started");
  console.log("---------------------------");
 
  // Keep the process alive until the browser is closed
  await new Promise((resolve) => {
    browser.on("disconnected", () => {
      resolve();
    });
  });
})();
 